src/librustdoc/lint.rs

```diff
@@ -204,6 +204,20 @@ declare_rustdoc_lint! {
     "detects markdown that is interpreted differently in different parser"
 }
 
+declare_rustdoc_lint! {
+    /// This lint checks for uses of footnote references without definition.
+    BROKEN_FOOTNOTE,
+    Warn,
+    "footnote reference with no associated definition"
+}
+
+declare_rustdoc_lint! {
+    /// This lint checks if all footnote definitions are used.
+    UNUSED_FOOTNOTE_DEFINITION,
+    Warn,
+    "unused footnote definition"
```

Comment:
```suggestion
    "detects footnote references with no associated definition"
}

declare_rustdoc_lint! {
    /// This lint checks if all footnote definitions are used.
    UNUSED_FOOTNOTE_DEFINITION,
    Warn,
    "detects unused footnote definitions"
```

for consistency with other lint descriptions